haystack/preview/components/builders/answer_builder.py

```diff
@@ -42,7 +42,7 @@ def run(
         self,
         query: str,
         replies: List[str],
-        metadata: List[Dict[str, Any]],
+        metadata: Optional[List[Dict[str, Any]]] = None,
```

Comment:
Let's also update the docstring for `:param metadata:` with something like "If not specified..."